src/main/java/io/github/thebusybiscuit/slimefun4/core/attributes/CustomProtection.java

```diff
@@ -0,0 +1,41 @@
+package io.github.thebusybiscuit.slimefun4.core.attributes;
+
+import org.bukkit.entity.Player;
+import org.bukkit.event.entity.EntityDamageEvent;
+
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.HazmatArmorPiece;
+import io.github.thebusybiscuit.slimefun4.implementation.items.armor.SlimefunArmorPiece;
+
+/**
+ * Implement this interface for any {@link SlimefunArmorPiece} to prevent
+ * the {@link Player} wearing that {@link SlimefunArmorPiece}
+ *
+ * <b>Important</b>: This will not cancel any {@link EntityDamageEvent}.
+ * It will simply prevent Slimefun from ever applying {@link ProtectionType}
+ * to this {@link SlimefunArmorPiece}'s wearer.
```

Comment:
I don't understand what this class does at all from this comment. It doesn't let people put on this armour? But also (?) stops "applying protection", what does that mean?

This really needs to be reworded.